driver-core/src/test/unit/com/mongodb/internal/connection/TestServerListener.java

```diff
@@ -23,31 +23,47 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.Predicate;
 
-import static com.mongodb.assertions.Assertions.isTrue;
 import static com.mongodb.assertions.Assertions.notNull;
 
-class TestServerListener implements ServerListener {
+public class TestServerListener implements ServerListener {
     private ServerOpeningEvent serverOpeningEvent;
     private ServerClosedEvent serverClosedEvent;
     private final List<ServerDescriptionChangedEvent> serverDescriptionChangedEvents = new ArrayList<ServerDescriptionChangedEvent>();
+    private final Lock lock = new ReentrantLock();
+    private final Condition condition = lock.newCondition();
+    private volatile int waitingForEventCount;
+    private Predicate<ServerDescriptionChangedEvent> waitingForEventMatcher;
 
     @Override
     public void serverOpening(final ServerOpeningEvent event) {
-        isTrue("serverOpeningEvent is null", serverOpeningEvent == null);
```

Comment:
These assertions make it impossible to use this class in a context where there is more than one server (e.g. a replica set).  It seemed fine to remove them.

Reply:
Ack